rollup.config.js

```diff
@@ -63,15 +39,49 @@ export default {
     }),
     sizeSnapshot(),
   ],
-  external: [
-    '@emotion/core',
-    '@emotion/styled',
-    'prop-types',
-    'react',
-    'react-modal',
-    'react-slick',
-    'react-toggle-button',
-    'react-transition-group',
-    'tinycolor2',
-  ],
+  external: [...Object.keys(pkg.dependencies), '@emotion/styled-base'],
```

Comment:
This should be a bit more sustainable.  We're telling rollup that all of the dependencies will be installed alongside this package.  This prevents the problem of duplicating `react-dom` within the package for the other versions.  Also `@emotion/styled-base` is necessary in addition because `Rollup` effectively bypasses some intermediate imports from `@emotion/styled` to pull directly from `@emotion/styled`.

Reply:
With `preserveModules` it will try to create a `node_modules` folder in the dist and then import from there (with a relative path).  But those get filtered out in the publish.  Telling rollup that these are all external lets rollup know that it should be able to just use `import ReactDom from 'react-dom'` and that will be available.

Reply:
`Also @emotion/styled-base is necessary in addition because Rollup effectively bypasses some intermediate imports from @emotion/styled to pull directly from @emotion/styled.` Can we memorialize this as a code comment?

Reply:
@timmartin19 Is your sense that including `react` but not `react-dom` before was an oversight?

Reply:
Yeah, I think we just started using react-dom for the portal function in the newer dialogs.  I think we already had `react-dom` in the dev dependencies but it wasn't strictly necessary for production code until we added the portal.